<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"><head><title>Async_unix__Reader (async_unix.Async_unix__Reader)</title><link rel="stylesheet" href="../../_odoc-theme/odoc.css"/><meta charset="utf-8"/><meta name="generator" content="odoc v1.1.1-1373-g2e7c1c75"/><meta name="viewport" content="width=device-width,initial-scale=1.0"/><script src="../../highlight.pack.js"></script><script>hljs.initHighlightingOnLoad();</script></head><body><nav><a href="../index.html">Up</a> â€“ <a href="../index.html">async_unix</a> &#x00BB; Async_unix__Reader</nav><header><h1>Module <code>Async_unix__Reader</code></h1><p><code>Reader</code> is Async's main API for buffered input from a file descriptor. It is the analog of <a href="../../stdio/Stdio/In_channel/index.html"><code>Stdio.In_channel</code></a>.</p><p>Each reader has an internal buffer, which is filled via <code>read()</code> system calls when data is needed to satisfy a <code>Reader.read*</code> call.</p><p>Each of the read functions returns a deferred that will become determined when the read completes. It is an error to have two simultaneous reads. That is, if you call a read function, you should not call another read function until the first one completes.</p><p>If the file descriptor underlying a reader is closed, the reader will return EOF (after all the buffered bytes have been read).</p><p>Any <code>Reader.read*</code> call could, rather than determine its result, send an exception to the monitor in effect when <code>read</code> was called. Such exceptions can be handled in the usual way by using <code>try_with</code>, e.g.:</p><pre><code>try_with (fun () -&gt; Reader.read reader ...) </code></pre></header><div class="content"></div></body></html>